test/Rx/Functional/FunctionalTestCase.php

```diff
@@ -99,4 +112,72 @@ protected function createTestScheduler()
     {
         return new TestScheduler();
     }
+
+    protected function convertMarblesToMessages(string $marbles, array $eventMap = [], \Exception $customError = null, $subscribePoint = 0)
+    {
+        var_dump($eventMap);
+        /** @var Recorded $events */
+        $events = [];
+        $zero = 0;
+
+        for ($i = 0; $i < strlen($marbles); $i++) {
+            switch ($marbles[$i]) {
+                case '-': // nothing
+                    continue;
+                case '#': // error
+                    $events[] = onError($i * 10, $customError ?? new \Exception());
+                    continue;
+                case '^': // this is the subscribe point
+                    $zero = $i * 10;
+                    continue;
+                case '|': //
+                    $events[] = onCompleted($i * 10);
+                    continue;
+                default:
+                    $events[] = onNext($i * 10, isset($eventMap[$i]) ? $eventMap[$i] : $marbles[$i]);
```

Comment:
Just `$eventMap[$i]` assumes that the `$eventMap` array is indexed by time (eg. `[0 => 'abc', 13 => 'def', 16 => ...]`)

For #144 I updated it to use the same approach as RxJS 5:

    $eventKey = $marbles[$i];
    $events[] = onNext($i * 10, isset($eventMap[$eventKey]) ? $eventMap[$eventKey] : $marbles[$i]);

So I can use it as `[ 'x' => $x, 'y' => $y, 'z' => $z ]`

https://github.com/martinsik/RxPHP/blob/002671800236c56d872a68dc635a5836c4f39448/test/Rx/Functional/FunctionalTestCase.php#L169